### ISSUE TITLE
Error while dialing dial unix /run/nordvpnd.sock: connect: no such file or directory

### ISSUE THREAD

**Issue by jackwilsdon** [bubuntux/nordvpn#80]
**Describe the bug**
The container fails to start, reporting the error `all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial unix /run/nordvpnd.sock: connect: no such file or directory"`.

**To Reproduce using docker CLI**
This is the command run by Unraid to start the container:

```Shell
docker run \
  -d \
  --name='vpn' \
  --net='bridge' \
  -e TZ="Europe/London" \
  -e HOST_OS="Unraid" \
  -e 'USER'='...' \
  -e 'PASS'='...' \
  -e 'TECHNOLOGY'='NordLynx' \
  -e 'NETWORK'='10.10.10.0/24' \
  -e 'DNS'='1.1.1.1,1.0.0.1' \
  -e 'DEBUG'='on' \
  --device='/dev/net/tun' \
  --cap-add=NET_ADMIN \
  --cap-add=SYS_MODULE \
  'bubuntux/nordvpn' 
```

**Expected behavior**

**Logs**
```Text
+ [[ -n '' ]]
+ [[ '' =~ ^[0-9]+$ ]]
++ ip -o addr show dev eth0
++ awk '$3 == "inet" {print $4}'
+ DOCKER_NET=172.17.0.2/16
+ kill_switch
++ ip -o addr show dev eth0
++ awk '$3 == "inet6" {print $4; exit}'
+ local docker6_network=
+ iptables -F
+ iptables -X
+ iptables -P INPUT DROP
+ iptables -P FORWARD DROP
+ iptables -P OUTPUT DROP
+ iptables -A INPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
+ iptables -A INPUT -i lo -j ACCEPT
+ iptables -A FORWARD -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
+ iptables -A FORWARD -i lo -j ACCEPT
+ iptables -A OUTPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
+ iptables -A OUTPUT -o lo -j ACCEPT
+ iptables -A OUTPUT -o tap+ -j ACCEPT
+ iptables -A OUTPUT -o tun+ -j ACCEPT
+ iptables -A OUTPUT -m owner --gid-owner vpn -j ACCEPT
+ iptables -t nat -A POSTROUTING -o tap+ -j MASQUERADE
+ iptables -t nat -A POSTROUTING -o tun+ -j MASQUERADE
+ [[ -n 172.17.0.2/16 ]]
+ iptables -A INPUT -s 172.17.0.2/16 -j ACCEPT
+ iptables -A FORWARD -d 172.17.0.2/16 -j ACCEPT
+ iptables -A FORWARD -s 172.17.0.2/16 -j ACCEPT
+ iptables -A OUTPUT -d 172.17.0.2/16 -j ACCEPT
+ [[ -n 10.10.10.0/24 ]]
+ for net in ${NETWORK//[;,]/ }
+ return_route 10.10.10.0/24
++ ip route
++ awk '/default/ {print $3}'
+ local network=10.10.10.0/24 gw=172.17.0.1
+ grep -q 10.10.10.0/24
+ ip route
+ ip route add to 10.10.10.0/24 via 172.17.0.1 dev eth0
+ iptables -A INPUT -s 10.10.10.0/24 -j ACCEPT
+ iptables -A FORWARD -d 10.10.10.0/24 -j ACCEPT
+ iptables -A FORWARD -s 10.10.10.0/24 -j ACCEPT
+ iptables -A OUTPUT -d 10.10.10.0/24 -j ACCEPT
+ [[ -n '' ]]
+ ip6tables -F
+ ip6tables -X
+ ip6tables -P INPUT DROP
+ ip6tables -P FORWARD DROP
+ ip6tables -P OUTPUT DROP
+ ip6tables -A INPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
+ ip6tables -A INPUT -p icmp -j ACCEPT
+ ip6tables -A INPUT -i lo -j ACCEPT
+ ip6tables -A FORWARD -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
+ ip6tables -A FORWARD -p icmp -j ACCEPT
+ ip6tables -A FORWARD -i lo -j ACCEPT
+ ip6tables -A OUTPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
+ ip6tables -A OUTPUT -o lo -j ACCEPT
+ ip6tables -A OUTPUT -o tap+ -j ACCEPT
+ ip6tables -A OUTPUT -o tun+ -j ACCEPT
+ ip6tables -A OUTPUT -m owner --gid-owner vpn -j ACCEPT
+ [[ -n '' ]]
+ [[ -n '' ]]
+ pkill nordvpnd
+ rm -f /run/nordvpnd.sock
+ sleep 0.5
+ sg vpn -c nordvpnd
+ nordvpn login -u ... -p ...
+ setup_nordvpn
+ [[ -n NordLynx ]]
+ nordvpn set technology NordLynx
all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial unix /run/nordvpnd.sock: connect: no such file or directory"
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n 1.1.1.1,1.0.0.1 ]]
+ nordvpn set dns 1.1.1.1 1.0.0.1
all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial unix /run/nordvpnd.sock: connect: no such file or directory"
+ [[ -n 172.17.0.2/16 ]]
+ nordvpn whitelist add subnet 172.17.0.2/16
all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial unix /run/nordvpnd.sock: connect: no such file or directory"
+ [[ -n 10.10.10.0/24 ]]
+ for net in ${NETWORK//[;,]/ }
+ nordvpn whitelist add subnet 10.10.10.0/24
all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial unix /run/nordvpnd.sock: connect: no such file or directory"
+ [[ -n '' ]]
+ [[ -n on ]]
+ nordvpn -version
NordVPN Version 3.7.2
+ nordvpn settings
+ create_tun_device
+ mkdir -p /dev/net
+ [[ -c /dev/net/tun ]]
+ nordvpn connect
Stdin/Stdout should be terminal
```

**Additional context**
This doesn't seem to happen every time when starting the container - it has started fine with the same configuration in the past.

**Comment by jackwilsdon** [comment]
Just tried again and the container started properly this time - could this be a race condition between starting `nordvpnd` and issuing `nordvpn` commands? I see that the container only waits half a second for the daemon to start:

https://github.com/bubuntux/nordvpn/blob/a081f8a97c12e553b0a60fe4722ef20130692e5d/start_vpn.sh#L120-L121

My machine is under high CPU and I/O load at the moment, so starting the daemon may be slower than usual.

Could we maybe add a loop here to wait for the socket to appear?

**Comment by bubuntux** [comment]
yes that would be an acceptable solution

**Comment by bubuntux** [comment]
Closed by #81
